docs/architecture/adr-003-subaccounts.md

```diff
@@ -0,0 +1,88 @@
+# ADR 3: Module SubAccounts
+
+## Changelog
+
+## Context
+
+Currently `ModuleAccount`s must be declared upon supply keeper initialization. In addition to this they don't allow for separation of fungible coins within an account.
+
+New structs should be added so a `ModuleAccount` can dynamically add accounts.
+
+## Decision
+
+Add the following structs into `x/supply`.
+
+### Implementation Changes
+
+Introduce two new structs:
+
+* `ModuleMultiAccount`
```

Comment:
What are your thoughts on revising the nomenclature here? Thoughts on `ModuleMultiAccount` and `ModuleAccount`. The former may have one or more of the latter.

Reply:
I like this, didn't even think about having `SubAccount` just be the existing ModuleAccount